### DISCUSSION TITLE
多实例是如何使用？

### DISCUSSION THREAD

**Discussion by leecoooo** [eric2788/vup_monitors#36] [Q&A] [answered]
},
    // websocket 数据源
    // 使用 blive.ericlamm.xyz 可贡献统计数据
    "websocket": {
        "id": "vup_monitors", // 支持不同ID以支援多个实例
        "host": "blive.ericlamm.xyz",
        "use-tls": true
    },

是可以添加多个websocket数据源的意思吗？ 如果是自架是如何开启多个ws？还是多开gc绑定多个QQ？
实例使用指令的时候是如何使用到以上的 "id": "vup_monitors"？

**Comment by eric2788** [comment, marked as answer]
多实例就是支援不同的直播监听订阅列表，在biligo-live-ws监听订阅区分使用ip@ID来区分，也就是说如果没有指明ID，那么将是一个IP共享一个订阅列表

**Reply by leecoooo** [reply]
> 多实例就是支援不同的直播监听订阅列表，在biligo-live-ws监听订阅区分使用ip@ID来区分，也就是说如果没有指明ID，那么将是一个IP共享一个订阅列表

还是不太理解能不能给个示例？

**Reply by eric2788** [reply]
简单来说就是开两个vup monitors，然后用上不同id，这样订阅列表就不会共享

**Comment by Marxel-cn** [comment]
> ```
> },
> // websocket 数据源
> // 使用 blive.ericlamm.xyz 可贡献统计数据
> "websocket": {
>     "id": "vup_monitors", // 支持不同ID以支援多个实例
>     "host": "blive.ericlamm.xyz",
>     "use-tls": true
> },
> ```
> 
> 是可以添加多个websocket数据源的意思吗？ 如果是自架是如何开启多个ws？还是多开gc绑定多个QQ？ 实例使用指令的时候是如何使用到以上的 "id": "vup_monitors"？

@leecoooo 问下这个还需要么？我准备给个PR来支持多实例命令，需要的话回头可以帮忙验证一下么？

**Comment by Marxel-cn** [comment]
@leecoooo 可以试下这个
https://github.com/eric2788/vup_monitors/suites/11494201255/artifacts/593821129
使用方法可以参看develop下面的README
https://github.com/eric2788/vup_monitors/tree/develop

**Reply by leecoooo** [reply]
收到，我有空测试一下

**Reply by leecoooo** [reply]
只设定"identifier": "",的时候，会有bw冲突，两个实例都3分钟重连一次。
同时也设定"id": " ",就不会断线了。

**Reply by leecoooo** [reply]
新发现一个问题，频道新增高亮的时候同时候也会增加一条私聊高亮，应该是判断出了问题。

**Reply by Marxel-cn** [reply]
> 只设定"identifier": "",的时候，会有bw冲突，两个实例都3分钟重连一次。 同时也设定"id": " ",就不会断线了。

嗯，共用bw的时候要在websocket里设置不同的id，共用qq的时候要设置不同的identifier，两个是独立的

**Reply by Marxel-cn** [reply]
> 新发现一个问题，频道新增高亮的时候同时候也会增加一条私聊高亮，应该是判断出了问题。

是highlight么？用的什么样的命令？通过私聊新增群（不是频道）高亮也有同样的问题么？
我自己测试没有复现这个问题

**Reply by leecoooo** [reply]
> > 新发现一个问题，频道新增高亮的时候同时候也会增加一条私聊高亮，应该是判断出了问题。
> 
> 是highlight么？用的什么样的命令？通过私聊新增群（不是频道）高亮也有同样的问题么？ 我自己测试没有复现这个问题

比如这样：!高亮 新增 xxxxxxxx 71087781643123732-238125789 #xx

是私聊添加，还有个可能是已经在列表中已经有了，重复添加的时候就会出现。

**Reply by Marxel-cn** [reply]
> !高亮 新增 xxxxxxxx 71087781643123732-238125789 #xx

![image](https://user-images.githubusercontent.com/69199776/226232821-d84ea7e1-5511-4696-a422-72a7b45408ee.png)
还是没有复现出来

**Reply by leecoooo** [reply]
> > !高亮 新增 xxxxxxxx 71087781643123732-238125789 #xx
> 
> ![image](https://user-images.githubusercontent.com/69199776/226232821-d84ea7e1-5511-4696-a422-72a7b45408ee.png) 还是没有复现出来

我重复操作了暂时也复现不出来，先忽略吧

**Reply by Marxel-cn** [reply]
> > > !高亮 新增 xxxxxxxx 71087781643123732-238125789 #xx
> > 
> > 
> > ![image](https://user-images.githubusercontent.com/69199776/226232821-d84ea7e1-5511-4696-a422-72a7b45408ee.png) 还是没有复现出来
> 
> 我重复操作了暂时也复现不出来，先忽略吧

复现的话另外提issue吧，这个我先关了
